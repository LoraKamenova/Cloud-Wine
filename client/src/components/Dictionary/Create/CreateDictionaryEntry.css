.add-dictionary-entry-form-wrapper > .form-header-wrapper > .form-header-heading-container > .form-header-heading {
    top: 30px;
    left: 25px;
}
