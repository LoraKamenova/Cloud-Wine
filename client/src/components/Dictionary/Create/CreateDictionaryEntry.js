import './CreateDictionaryEntry.css'
import FormHeader from "../../Shared/FormHeader/FormHeader";
import InputError from "../../Shared/InputError/InputError";

import * as dictionaryService from "../../../services/dictionaryService";

import {useHistory} from "react-router-dom"
import React, {useState} from "react";

const CreateDictionaryEntry = () => {
    const history = useHistory();
    const [errorMessage, setErrorMessage] = useState('');

    const onCreateDictionaryEntrySubmitHandler = (e) => {
        e.preventDefault();

        const {
            title,
            fullText,
            shortText,
            iconWhiteUrl,
            iconBlackUrl,
            imageUrl,
        } = e.target;

        dictionaryService.create(
            title.value,
            fullText.value,
            shortText.value,
            iconWhiteUrl.value,
            iconBlackUrl.value,
            imageUrl.value,
         )
            .then(() => {
                history.push('/dictionary/all')
            })
    };

    const onChangeShortTextHandler = (e) => {
        if (e.target.value.length > 80) {
            setErrorMessage('Текстът трябва да е максимум 80 символа');
        } else {
            setErrorMessage('');
        }
    };

    return (
        <section className="form-wrapper add-dictionary-entry-form-wrapper">
            <FormHeader title="Нов термин"/>

            <form className="form-container" onSubmit={onCreateDictionaryEntrySubmitHandler}>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="title">Заглавие:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="title"
                           name="title"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="fullText">Пълен текст:</label>
                    </div>
                    <textarea className="form-field-input textarea"
                              rows="11"
                              id="fullText"
                              name="fullText"
                              required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="shortText">Кратък текст:</label>
                    </div>
                    <textarea className="form-field-input textarea"
                              rows="2"
                              id="shortText"
                              name="shortText"
                              required
                              onBlur={onChangeShortTextHandler}/>
                    <InputError>{errorMessage}</InputError>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="iconBlackUrl">URL черна иконка:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="iconBlackUrl"
                           name="iconBlackUrl"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="iconWhiteUrl">URL бяла икона:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="iconWhiteUrl"
                           name="iconWhiteUrl"
                           required/>
                </div>

                <div className="form-field-wrapper">
                    <div className="form-field-heading">
                        <label htmlFor="imageUrl">URL снимка:</label>
                    </div>
                    <input className="form-field-input"
                           type="text"
                           id="imageUrl"
                           name="imageUrl"
                           required/>
                </div>

                <div className="form-button-wrapper">
                    <input className="form-button" type="submit" value="ИЗПРАТИ"/>
                </div>

            </form>
        </section>
    );
};

export default CreateDictionaryEntry;
