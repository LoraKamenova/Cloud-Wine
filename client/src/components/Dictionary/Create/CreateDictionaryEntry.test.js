import { render } from '@testing-library/react';
import CreateDictionaryEntry from './CreateDictionaryEntry';

describe('CreateDictionaryEntry Component', () => {

    it("CreateDictionaryEntry Component snapshot", () => {
        const { asFragment } = render(<CreateDictionaryEntry />);
        expect(asFragment()).toMatchSnapshot();
    });

});
