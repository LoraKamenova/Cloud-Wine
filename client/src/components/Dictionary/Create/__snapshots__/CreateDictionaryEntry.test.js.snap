// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDictionaryEntry Component CreateDictionaryEntry Component snapshot 1`] = `
<DocumentFragment>
  <section
    class="form-wrapper add-dictionary-entry-form-wrapper"
  >
    <div
      class="form-header-wrapper"
    >
      <div
        class="form-header-heading-container"
      >
        <h1
          class="form-header-heading"
        >
          Нов термин
        </h1>
      </div>
      <div
        class="form-header-image-container"
      >
        <img
          alt="vine"
          class="form-header-image"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1615712293/React%20Project/grape_border_ogd8gq.png"
        />
      </div>
      <h1
        class="form-header-heading-hidden"
      >
        Нов термин
      </h1>
    </div>
    <form
      class="form-container"
    >
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="title"
          >
            Заглавие:
          </label>
        </div>
        <input
          class="form-field-input"
          id="title"
          name="title"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="fullText"
          >
            Пълен текст:
          </label>
        </div>
        <textarea
          class="form-field-input textarea"
          id="fullText"
          name="fullText"
          required=""
          rows="11"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="shortText"
          >
            Кратък текст:
          </label>
        </div>
        <textarea
          class="form-field-input textarea"
          id="shortText"
          name="shortText"
          required=""
          rows="2"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="iconBlackUrl"
          >
            URL черна иконка:
          </label>
        </div>
        <input
          class="form-field-input"
          id="iconBlackUrl"
          name="iconBlackUrl"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="iconWhiteUrl"
          >
            URL бяла икона:
          </label>
        </div>
        <input
          class="form-field-input"
          id="iconWhiteUrl"
          name="iconWhiteUrl"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="imageUrl"
          >
            URL снимка:
          </label>
        </div>
        <input
          class="form-field-input"
          id="imageUrl"
          name="imageUrl"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-button-wrapper"
      >
        <input
          class="form-button"
          type="submit"
          value="ИЗПРАТИ"
        />
      </div>
    </form>
  </section>
</DocumentFragment>
`;
